Modules/booking/Controller/BookingdefaultController.php

```diff
@@ -293,31 +284,21 @@ public function editreservationqueryAction($id_space) {
             }
         } else {
             $periodicEndDate = CoreTranslator::dateToEn($this->request->getParameter("periodic_enddate"), $lang);
-            $periodicEndDateArray = explode("-", $periodicEndDate);
-            $last_start_time = mktime($hour_startH, $hour_startM, 0, $periodicEndDateArray[1], $periodicEndDateArray[2], $periodicEndDateArray[0]);
-
+            $periodicEndDateArray = explode("/", $periodicEndDate);
```

Comment:
separator is different here, dateToEn returns with "-" or "/" ????  how is it done with other dateToEn calls?

Reply:
dateToEn, if lang=en, will return date, but if lang=fr will return YYYY-MM-DD

so i suppose you want to explode original date, not translated date (I am not sure with parameter gives according to lang)

    $periodicEndDate = CoreTranslator::dateToEn($this->request->getParameter("periodic_enddate"), $lang);
    $periodicEndDateArray = explode("/", $this->request->getParameter("periodic_enddate"));

Reply:
in addreservationaction , dates taken as:

    $('.form_date_fr').datetimepicker({
                    language: 'fr',
                    weekStart: 1,
                    todayBtn: 1,
                    autoclose: 1,
                    todayHighlight: 1,
                    startView: 2,
                    minView: 2,
                    forceParse: 0,
                    format: 'dd/mm/yyyy'
                });
                $('.form_date_en').datetimepicker({
                    language: 'us',
                    weekStart: 1,
                    todayBtn: 1,
                    autoclose: 1,
                    todayHighlight: 1,
                    startView: 2,
                    minView: 2,
                    forceParse: 0,
                    format: 'yyyy-mm-dd'
                });


so different format according to lang, though in code only form_date_fr is set... so we can suppose we always receive (at least here) dd/mm/yyyy so dateToEn will return YYY-MM-DD

In english, will also have YYY-MM-DD, so should indeed keep original   $periodicEndDateArray = explode("-", $periodicEndDate)

Reply:
Thanks for having brought that to my attention. I went too fast on this. The original code wasn't working for me. I logged the date and it was including "/", not "-". I gonna take language into account to make it work in both case

Reply:
It appeared that date format isn't always linked to $lang (at least in bookingdefaultController) and, consequently, if $lang="en" and date format is dd/mm/yyyy, dateToEn() in CoreTranslator.php will return a date in french format and an error will be risen.
It is possible that this case stands in multiple places, so, for a short-term efficient fix, I replaced in dateToEn($date, $lang) ```if ($lang == "fr")``` by ```if ($lang == "fr" || str_contains($date, "/"))```

Reply:
see issue #200